packages/kbn-esql-utils/src/utils/query_parsing_helpers.ts

```diff
@@ -136,3 +136,14 @@ export const retrieveMetadataColumns = (esql: string): string[] => {
   const metadataOptions = options.find(({ name }) => name === 'metadata');
   return metadataOptions?.args.map((column) => (column as ESQLColumn).name) ?? [];
 };
+
+export const getQueryColumnsFromESQLQuery = (esql: string): string[] => {
+  const { root } = parse(esql);
+  const columns: ESQLColumn[] = [];
+
+  walk(root, {
+    visitColumn: (node) => columns.push(node),
```

Comment:
Will this visit variables as well?

Reply:
if you see the examples yes.

Reply:
Ok, only reason I ask is that it seemed like this was connected to the request for field use telemetry. But, I guess you'll filter out the variables as a second step.

Reply:
This might be ok anyway depending on how we think about it (open to thoughts). We'd filter out non-ECS field names before sending them since custom names could include PII, although we may replace them with a placeholder like `__CUSTOM_FIELD__` in telemetry. If a user runs a query with `EVAL log.level = ...` then the field would get included as an ECS field, but maybe that makes sense if they're generating a column that matches our schema?

Reply:
Yes I think that Discover will already try to check for ECS fields so I dont want to change the helper here as it might be in handl in other use cases too that the variables are important

Reply:
About:

>  but maybe that makes sense if they're generating a column that matches our schema?

yes exactly ++